front/lib/resources/data_source_resource.ts

```diff
@@ -362,6 +366,31 @@ export class DataSourceResource extends ResourceWithVault<DataSourceModel> {
     auth: Authenticator,
     transaction?: Transaction
   ): Promise<Result<undefined, Error>> {
+    try {
+      await this.model.destroy({
+        where: {
+          id: this.id,
+        },
+        transaction,
+      });
+
+      // We directly delete the DataSourceViewResource.model here to avoid a circular dependency.
+      // DataSourceViewResource depends on DataSourceResource
+      await DataSourceViewModel.destroy({
+        where: {
+          workspaceId: auth.getNonNullableWorkspace().id,
+          dataSourceId: this.id,
+        },
+        transaction,
+      });
+
+      return new Ok(undefined);
+    } catch (err) {
+      return new Err(err as Error);
+    }
+  }
+
+  async destroy(auth: Authenticator, transaction?: Transaction): Promise<void> {
```

Comment:
this is a bit ill-named IMHO I have a hard time tracking the delete vs the model delete vs destroy tbh.

I think we should avoid adding a new verb if we can and be more explcit. Maybe delete with a hardDelete option vs delete/destroy?